### ISSUE TITLE
Incorrect handling of quotes in lineinfile module

### ISSUE THREAD

**Issue by devoto13** [ansible/ansible-modules-core#2227]
### Issue Type

Bug Report
### Component Name

lineinfile module
### Ansible Version

1.9.3
### Summary

I am using Ansible 1.9.3.

I have following content in my `test.yml` file:

``` yaml

---
- hosts: all
  connection: local
  tasks:
    - lineinfile:
        dest: ./example.txt
        create: yes
        line: "something '\"content\"' something else"
```

And I run it as follows:

`ansible-playbook -i localhost, test.yml`
### Expected result

`example.txt` contains `something '"content"' something else`.
### Actual result

`example.txt` contains `something content something else`. Without any quotes.
### Investigation

It seems the problem lies in the `lineinfile` module adding quotes and then executing `line = module.safe_eval(line)`. Relevant [code](https://github.com/ansible/ansible-modules-core/blob/cf88f2786822ab5f4a1cd711761a40df49bd93f0/files/lineinfile.py). 

After adding quotes line looks like `'something '"content"' something else'` and when passed to `module.safe_eval()` the Pythons's implicit string concatenation is applied and it looses all quotes.

---

**Haven't checked if problem exists in development version, it seems module was significantly rewritten.**

**Comment by robynbergeron** [comment]
[module: files/lineinfile.py]

**Comment by robynbergeron** [comment]
@devoto13 -- Thanks for submitting this bug report! Much appreciated. 

@dhozac & @ahtik -- thanks in advance for taking a look at this bug report and for your ongoing work in maintaining this module.

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by silverdr** [comment]
Has this actually been fixed?

**Comment by dagwieers** [comment]
Easy to test, no ?

**Comment by silverdr** [comment]
Actually yes. Still - it would be nice to see a conclusive/informative closing like "Fixed in #...." or "wontfix", etc.

**Comment by devoto13** [comment]
I stopped using Ansible long ago and since this issue didn't get any attention in more than a year I decided that it is fixed in one of the newer versions and closed the issue.

If it is still not fixed I have described workaround in [this answer](https://stackoverflow.com/a/33019626/1377864) on StackOverflow.

**Comment by dagwieers** [comment]
@silverdr It's the original reporter that closed it. You can open a new ticket at https://github.com/ansible/ansible/issues/new
